tests/test_db_template_CRUD.py

```diff
@@ -0,0 +1,443 @@
+    assert new_page.controller == new_page_config["controller"]
+
+def test_update_page(flask_test_client, _db, clear_test_data, seed_dynamic_data):
+    new_page_config["form_id"] = None
+    new_page = insert_new_page(new_page_config)
+
+    assert new_page.form_id is None
+    assert new_page.name_in_apply_json == new_page_config["name_in_apply_json"]
+    assert new_page.template_name == None
+    assert new_page.is_template == False
+    assert new_page.source_template_id == None
+    assert new_page.audit_info == new_page_config["audit_info"]
+    assert new_page.form_index == new_page_config["form_index"]
+    assert new_page.display_path == new_page_config["display_path"]
+    assert new_page.controller == new_page_config["controller"]
+
+    # Update new_page_config
+    updated_page_config = deepcopy(new_page_config)
+    updated_page_config['name_in_apply_json'] = {"en": "Updated Page Name"}
+    updated_page_config['audit_info'] = {"created_by": "Jonny Doe", "created_at": "2024-01-02"}
+
+    updated_page = update_page(new_page.page_id, updated_page_config)
+    
+    assert isinstance(updated_page, Page)
+    assert updated_page.form_id == updated_page_config["form_id"]
+    assert updated_page.name_in_apply_json == updated_page_config["name_in_apply_json"]
+    assert updated_page.audit_info == updated_page_config["audit_info"]
+
+def test_delete_page(flask_test_client, _db, clear_test_data, seed_dynamic_data):
+    new_page_config["form_id"] = None
+    new_page = insert_new_page(new_page_config)
+
+    assert isinstance(new_page, Page)
+    assert new_page.audit_info == new_page_config["audit_info"]
+
+    delete_page(new_page.page_id)
+    assert _db.session.query(Page).filter(Page.page_id == new_page.page_id).one_or_none() == None
+
+new_component_config = {
+    "page_id": uuid.uuid4(),
+    "theme_id": uuid.uuid4(),
+    "title": "Component Title",
+    "hint_text": "Component Hint Text",
+    "options": {"hideTitle": False, "classes": "test-class"},
+    "type": ComponentType.TEXT_FIELD,
+    "is_template": False,
+    "template_name": None,
+    "source_template_id": None,
+    "audit_info": {"created_by": "John Doe", "created_at": "2022-01-01"},
+    "page_index": 1,
+    "theme_index": 1,
+    "conditions": [
+            {
+                "name": "organisation_other_names_no",
+                "value": "false",  # this must be lowercaes or the navigation doesn't work
+                "operator": "is",
+                "destination_page_path": "CONTINUE",
+            },
+            {
+                "name": "organisation_other_names_yes",
+                "value": "true",  # this must be lowercaes or the navigation doesn't work
+                "operator": "is",
+                "destination_page_path": "organisation-alternative-names",
+            },
+        ],
+    "runner_component_name": "test-component",
+    "list_id": uuid.uuid4()
+}
+
+
+new_template_component_config = {
```

Comment:
shall we add these into a different file and we'll try to limit the code lines in a file if possible

Reply:
i kept them in file on purpose because in the past we have moved config out of file nad it gets very confusing, happy to chat this over with the devs and agree on a better approach